### PR TITLE
Added Series.sum feature for String

### LINKED ISSUE #3771: [FEA] Reductions over string columns
**Is your feature request related to a problem? Please describe.**
Pandas allows for certain types of reductions over string based series objects. In particular one can sum a `string` series and obtain an ordered concatenation of all the elements:

```
>>> pd.Series(['a','b','c']).sum()
'abc'
```

As well as perform `min` and `max` operations that seem to return alphabetical results:

```
>>> pd.Series(['a','b','c']).min()
'a'
```

It would be useful for matching pandas behavior around global reduction operations in cuDF such as `DataFrame.min()`, currently we have to drop some non-numeric columns to proceed in these cases. 

**Describe the solution you'd like**
Match pandas reductions over strings

**Describe alternatives you've considered**
I think a workaround might be possible by reusing some of the categorical machinery we have already. But ultimately native support would be optimal. 

**Additional context**
useful for https://github.com/rapidsai/cudf/issues/2067

### PR DESCRIPTION
**Pull request by tanyamehrotra01**
This PR implements the sum feature for Series with `string` type. The resulting behavior is the concatenation of all the strings given in the input series.

After completion, it closes #3771 

Unit Tests have also been added.

Example: 
```python
s = cudf.Series(['a','b','c'])
s.sum()

'abc'
```

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 - PR #3690 Add bools_to_mask
 - PR #3683 Added support for multiple delimiters in `nvtext.token_count()`
 - PR #3792 Adding is_nan and is_notnan
+- PR #3861 Added Series.sum feature for String
```
Comment (kkraus14) [inline CHANGELOG.md]: I've retargeted this PR to branch-0.13, could you move this to the proper changelog section?

**Comment by kkraus14** [comment]
ok to test

**Comment by kkraus14** [comment]
Thanks @tanyamehrotra01!

**Pull request merged**